### PR TITLE
falling on punji sticks now impales you

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Falling on punji sticks from multi-z impales you on them, doing a lot of damage, you need a few seconds to remove yourself from it.

## Why It's Good For The Game

image

## Changelog

:cl:
expansion: Falling on punji sticks impales you.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/towercap.dm [hunk 1]
```diff
@@ -136,10 +136,39 @@
...
+				var/mob/living/carbon/fallen_carbon = fallen_mob
+				fallen_carbon.emote("scream")
+				fallen_carbon.bleed(30)
+			add_overlay(stab_overlay)
+	. |= FALL_INTERCEPTED | FALL_NO_MESSAGE
+
+/obj/structure/punji_sticks/unbuckle_mob(mob/living/buckled_mob, force, can_fall)
+	if(force)
+		return ..()
+	to_chat(buckled_mob, span_warning("You begin climbing out of [src].</span>"))
```
Comment (MrMelbert) [inline code/modules/hydroponics/grown/towercap.dm:165]:
```suggestion
	to_chat(buckled_mob, span_warning("You begin climbing out of [src]."))
```

File: code/modules/hydroponics/grown/towercap.dm [hunk 2]
```diff
@@ -136,10 +136,39 @@
...
+	buckled_mob.apply_damage(5, BRUTE, sharpness = SHARP_POINTY)
+	if(do_after(buckled_mob, 5 SECONDS, target = src))
+		cut_overlay(stab_overlay)
+		return ..()
+	else
+		to_chat(buckled_mob, span_userdanger("You fail to detach yourself from [src]."))
```
Comment (MrMelbert) [inline code/modules/hydroponics/grown/towercap.dm:171]: Early return the doafter?

File: code/modules/hydroponics/grown/towercap.dm [hunk 3]
```diff
@@ -136,10 +136,39 @@
 	max_integrity = 30
 	density = FALSE
 	anchored = TRUE
+	buckle_lying = 90
+	var/mutable_appearance/stab_overlay
```
Comment (MrMelbert) [inline code/modules/hydroponics/grown/towercap.dm:140]: Auto-doc

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by datorangebottle** [comment]
finally, i can fill tram underpass with punji sticks and deliberately hit people with the tram so that they fucking die.

**Review by MrMelbert — APPROVED** [review]

**Pull request merged**